src/cloudabi-run/cloudabi-run.cc

```diff
@@ -129,13 +140,7 @@ int main(int argc, char *argv[]) {
     emulate(fd, buf.data(), buf.size(), &posix_syscalls);
```

Comment:
GitHub doesn't allow me to apply comments to pieces of code that are not modified? Anyway, the file descriptor passed to `emulate()` does not need to be opened separately. Change the `open()` call above to be something like:

    int execfd = open(argv[0], do_emulate ? O_RDONLY : O_EXEC);

You can then pass `execfd` to `emulate()`.